### PR TITLE
Issue #9142: Migrate to Truth in MutableExceptionCheckTest

### PR DESCRIPTION
**Pull request by shashwatj07**
Issue #9142

### CODE REVIEW

**Review by pbludov — APPROVED** [review]

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/design/MutableExceptionCheckTest.java [hunk 1]
```diff
@@ -19,9 +19,8 @@
 
 package com.puppycrawl.tools.checkstyle.checks.design;
 
+import static com.google.common.truth.Truth.assertWithMessage;
 import static com.puppycrawl.tools.checkstyle.checks.design.MutableExceptionCheck.MSG_KEY;
-import static org.junit.jupiter.api.Assertions.assertArrayEquals;
-import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.fail;
```
Comment (timurt) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/design/MutableExceptionCheckTest.java:24]: as discussed here https://github.com/checkstyle/checkstyle/issues/9142#issuecomment-855165188

please replace it with `assertWithMessage("message").fail();`

Comment (shashwatj07) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/design/MutableExceptionCheckTest.java]: done @timurt

**Comment by timurt** [comment]
Job pr-age fails because master is 12 commits ahead
rebase on latest master please

**Comment by shashwatj07** [comment]
> Job pr-age fails because master is 12 commits ahead
> rebase on latest master please

@timurt my bad. done.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/design/MutableExceptionCheckTest.java (hunk 1 above)
Comment (shashwatj07) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/design/MutableExceptionCheckTest.java]: @timurt 
```
[ERROR] src\test\java\com\puppycrawl\tools\checkstyle\checks\design\MutableExceptionCheckTest.java:[124,9] (extension) RequireFailForTryCatchInJunit: try/catch either requires a fail at the end of the try clause for junit tests or to remove the try/catch completely.
```

Comment (shashwatj07) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/design/MutableExceptionCheckTest.java]: Do we remove this check or suppress it?

Comment (timurt) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/design/MutableExceptionCheckTest.java]: lets suppress it via `sevnt_suppressions.xml`

Comment (shashwatj07) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/design/MutableExceptionCheckTest.java]: done

Comment (pbludov) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/design/MutableExceptionCheckTest.java]: > [ERROR] src\test\java\com\puppycrawl\tools\checkstyle\checks\design\MutableExceptionCheckTest.java:[124,9] (extension) RequireFailForTryCatchInJunit: try/catch either requires a fail at the end of the try clause for junit tests or to remove the try/catch completely.

This is actually a bug in `RequireFailForTryCatchInJunit`. There is a closed issue for this: https://github.com/sevntu-checkstyle/sevntu.checkstyle/issues/826. I'm going to investigate this issue.

File: config/sevntu_suppressions.xml [hunk 2]
```diff
@@ -40,4 +40,7 @@
   <!-- No need for constructor with cause -->
   <suppress checks="CauseParameterInException" files=".*MetadataGenerationException\.java"/>
 
+  <!-- Migrating to Truth -->
+  <suppress checks="RequireFailForTryCatchInJunit" files="MutableExceptionCheckTest.java"/>
```
Comment (pbludov) [inline config/sevntu_suppressions.xml:44]: This suppression is not needed.
The bug will be fixed in sevntu-checkstyle/sevntu.checkstyle#847

**Review by pbludov — CHANGES_REQUESTED** [review]
Please hold on till https://github.com/sevntu-checkstyle/sevntu.checkstyle/issues/847

File: config/sevntu_suppressions.xml (hunk 2 above)
Comment (timurt) [reply config/sevntu_suppressions.xml:44]: @shashwatj07 please subscribe to issue that @pbludov provided, to get notification when issue is closed

Comment (romani) [reply config/sevntu_suppressions.xml:44]: while bug is not fixed and released , we need to use suppression.

**Review by romani — APPROVED** [review]

File: config/sevntu_suppressions.xml (hunk 2 above)
Comment (romani) [reply config/sevntu_suppressions.xml:44]: @pbludov , please reconsider blocking such PRs, yes now it requires extra work, but it is not blocked and all suppressions could be removed ones sevntu issue is resolved.

Comment (pbludov) [reply config/sevntu_suppressions.xml:44]: sure, we can merge this first.

**Review by pbludov — APPROVED** [review]

**Comment by romani** [comment]
@shashwatj07 , please remove such suppressions ones sevntu issue is fixed.

**Pull request merged**